lib/mainFn/Host.ts

```diff
@@ -0,0 +1,185 @@
+import { Effects } from "../types"
+import { AddressReceipt } from "./AddressReceipt"
+import { NetworkInterfaceBuilder } from "./NetworkInterfaceBuilder"
+import { Origin } from "./Origin"
+
+const knownProtocols = {
+  http: {
+    secure: false,
+    ssl: false,
+    defaultPort: 80,
+    withSsl: "https",
+  },
+  https: {
+    secure: true,
+    ssl: true,
+    defaultPort: 443,
+  },
+  ws: {
+    secure: false,
+    ssl: false,
+    defaultPort: 80,
+    withSsl: "wss",
+  },
+  wss: {
+    secure: true,
+    ssl: true,
+    defaultPort: 443,
+  },
+  ssh: {
+    secure: true,
+    ssl: false,
+    defaultPort: 22,
+  },
+  bitcoin: {
+    secure: true,
+    ssl: false,
+    defaultPort: 8333,
+  },
+  grpc: {
+    secure: true,
+    ssl: true,
+    defaultPort: 50051,
+  },
+  dns: {
+    secure: true,
+    ssl: false,
+    defaultPort: 53,
+  },
+} as const
+
+type KnownProtocol = keyof typeof knownProtocols
+
+type Scheme = string | null
+
+type BasePortOptions<T extends KnownProtocol> = {
+  protocol: T
+  preferredExternalPort?: number
+  scheme?: Scheme
+}
+type AddSslOptions = {
+  preferredExternalPort: number
+  scheme: Scheme
+  addXForwardedHeaders?: boolean /** default: false */
```

Comment:
```suggestion
/** default: false */
  addXForwardedHeaders?: boolean 
```
This type should be before so the jsdoc can take advantage of